tests/core/CoreTest.py

```diff
@@ -514,7 +517,7 @@ def test_run_result_handler_exception(self):
         on_result = unittest.mock.Mock(side_effect=ValueError)
 
         with self.assertLogs(logging.getLogger()) as cm:
-            run({bear}, on_result)
+            self.execute_run({bear}, on_result)
```

Comment:
So, that every time, we just have to call `execute_run` even if the definition of run change in future.